File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int max = INT_MIN;
        int i=0;
        int j=n-1;
        while(i<j)
        {
            int area =  min(height[i],height[j])*(j-i);
            if(max<=area)
            {
                max = area;
            }
            if(height[i]<height[j])
            {
                i++;
            }
            else{
                j--;
            }

        }
        return max;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l=0,r=height.length-1;
        int maxArea = Math.abs(height[l]-height[r])*(l-r);
        while(l<r){
            maxArea = Math.max(Math.min(height[l],height[r])*(r-l), maxArea);
            if(height[l]<height[r])l++;
            else r--;
        }
        return maxArea;
    }
}